### DISCUSSION TITLE
AT!LTECA command always ERROR on EM7455

### DISCUSSION THREAD

**Discussion by pccr10001** [danielewood/sierra-wireless-modems#146] [General] [answered]
Hi there.

I have an EM7455 modem trying to enable LTE CA but `AT!LTECA?` and `AT!LTECA` command always failed.
I flashed 002.082 firmware and also done `AT!OPENLOCK` to make modem leave LPM mode, but I can't issue any `AT!LTECA` commands including querying and setting CA enabled.

LTE band supported by my ISP is B1+B3+B7+B8.

Here are the log
```
AT!ENTERCND="A710"
OK
AT!IMPREF?
!IMPREF:
 preferred fw version:    02.38.00.00
 preferred carrier name:  GENERIC
 preferred config name:   GENERIC_002.082_000
 preferred subpri index:  000
 current fw version:      02.38.00.00
 current carrier name:    GENERIC
 current config name:     GENERIC_002.082_000
 current subpri index:    000

OK
AT!GOBIIMPREF?
!GOBIIMPREF:
 preferred fw version:    02.38.00.00
 preferred carrier name:  GENERIC
 preferred config name:   GENERIC_002.082_000
 preferred subpri index:  000
 current fw version:      02.38.00.00
 current carrier name:    GENERIC
 current config name:     GENERIC_002.082_000
 current subpri index:    000

OK
AT!USBSPEED?
SUPPORTED:High-Speed
CURRENT  :High-Speed

OK
AT!USBCOMP?
Config Index: 1
Config Type:  1 (Generic)
Interface bitmask: 0000100D (diag,nmea,modem,mbim)

OK
AT!USBVID?
!USBVID:
1199

OK
AT!USBPID?
!USBPID:
APP : 9079
BOOT: 9078

OK
AT!USBPRODUCT?
Sierra Wireless EM7455 Qualcomm Snapdragon X7 LTE-A

OK
AT!PRIID?
PRI Part Number: 9906870
Revision: 001.000
Customer: Lenovo-Raven2

Carrier PRI: 9999999_9904609_SWI9X30C_02.38.00.00_00_GENERIC_002.082_000

OK
AT!SELRAT?
!SELRAT: 11, UMTS and LTE Only


OK
AT!BAND?
Index, Name,                        GW Band Mask     L Band Mask      TDS Band Mask
00, All bands,                      0002000007C00000 00000100030818DF 0000000000000000


OK
AT!PCINFO?
State: Online
LPM voters - Temp:0, Volt:0, User:0, W_DISABLE:0, IMSWITCH:0, BIOS:0, LWM2M:0, OMADM:0, FOTA:0
LPM persistence - None

OK
AT!PCOFFEN?
2

OK
AT!CUSTOM?
!CUSTOM:
             GPSENABLE          0x01
             GPSSEL             0x01
             IPV6ENABLE         0x01
             SIMLPM             0x01
             USBSERIALENABLE    0x01
             FASTENUMEN         0x02
             SINGLEAPNSWITCH    0x01


OK
AT!IMAGE?
TYPE SLOT STATUS LRU FAILURES UNIQUE_ID   BUILD_ID
FW   1    GOOD   1   0 0      ?_?         02.38.00.00_?
FW   2    EMPTY  0   0 0
FW   3    EMPTY  0   0 0
FW   4    EMPTY  0   0 0
Max FW images: 4
Active FW image is at slot 1

TYPE SLOT STATUS LRU FAILURES UNIQUE_ID   BUILD_ID
PRI  FF   GOOD   0   0 0      002.082_000 02.38.00.00_GENERIC
Max PRI images: 50


OK
AT!BAND=?
Index, Name,                        GW Band Mask     L Band Mask      TDS Band Mask
00, All bands                     0002000007C00000 00000100030818DF 0000000000000000
01, Europe 3G                     0002000000400000 0000000000000000 0000000000000000
02, North America 3G              0000000004800000 0000000000000000 0000000000000000
06, Europe                        0002000000400000 00000000000800C5 0000000000000000
07, North America                 0000000004800000 000000000300185A 0000000000000000
08, WCDMA ALL                     0002000007C00000 0000000000000000 0000000000000000
09, LTE ALL                       0000000000000000 00000100030818DF 0000000000000000

                                                   0000010000000000 - B41
                                                   0000000002000000 - B26
                                                   0000000001000000 - B25
                                                   0000000000080000 - B20
                                                   0000000000001000 - B13
                                                   0000000000000800 - B12
                                                   0000000000000080 - B8
                                                   0000000000000040 - B7
                                                   0000000000000010 - B5
                                                   0000000000000008 - B4
                                                   0000000000000004 - B3
                                                   0000000000000002 - B2
                                                   0000000000000001 - B1
                                  0002000000000000 - B8  (900)
                                  0000000004000000 - B5  (850)
                                  0000000002000000 - B4 (1700)
                                  0000000001000000 - B3 (1700)
                                  0000000000800000 - B2 (1900)
                                  0000000000400000 - B1 (2100)

OK
AT!GSTATUS?
!GSTATUS:
Current Time:  170              Temperature: 32
Reset Counter: 2                Mode:        ONLINE
System mode:   LTE              PS state:    Attached
LTE band:      B7               LTE bw:      20 MHz
LTE Rx chan:   3050             LTE Tx chan: 21050
LTE CA state:  NOT ASSIGNED
EMM state:     Registered       Normal Service
RRC state:     RRC Connected
IMS reg state: No Srv

PCC RxM RSSI:  -60              RSRP (dBm):  -93
PCC RxD RSSI:  -55              RSRP (dBm):  -88
Tx Power:      8                TAC:         5F50 (24400)
RSRQ (dB):     -11.5            Cell ID:     XXXXXXXX (XXXXXXXXX)
SINR (dB):     20.0


OK
AT!LTEINFO?
!LTEINFO:
Serving:   EARFCN MCC MNC   TAC      CID Bd D U SNR PCI  RSRQ   RSRP   RSSI RXLV
             3050 466  92 24400 XXXXXXXX  7 5 5  18  22  -7.6  -88.1  -60.2  35

IntraFreq:                                          PCI  RSRQ   RSRP   RSSI RXLV
                                                     22  -7.6  -88.1  -60.2  35

InterFreq: EARFCN ThresholdLow ThresholdHi Priority PCI  RSRQ   RSRP   RSSI RXLV

WCDMA:     UARFCN ThreshL ThreshH Prio PSC   RSCP  ECN0 RXLV


OK
AT!LTECA?
ERROR
AT!LTECA=?
!LTECA=<flag>
<flag>: 0 - disable CA
        1 - enable CA


OK
AT!LTECA
ERROR
AT!LTECA=1
OK
AT!LTECA?
ERROR
AT!LTECA=0
OK
AT!LTECA?
ERROR
```

Thank you.

**Comment by nicolae788** [comment]
Most likely you have to enable engineering mode with AT!OPENLOCK, before trying to modify that setting. -------- Original message --------From: PowerLi ***@***.***>Date: Sun, Feb 19, 2023, 05:02To: danielewood/sierra-wireless-modems ***@***.***>Cc: Subscribed ***@***.***>Subject: [danielewood/sierra-wireless-modems] AT!LTECA command always ERROR on EM7455 (Discussion #146)
Hi there.
I have an EM7455 modem trying to enable LTE CA but AT!LTECA? and AT!LTECA command always failed.
I flashed 002.082 firmware and also done AT!OPENLOCK to make modem leave LPM mode.
I can't issue any AT!LTECA commands including querying and setting CA enabled.
LTE band supported by my ISP is B1+B2+B7+B8.
Here are the log
AT!ENTERCND="A710"
OK
AT!IMPREF?
!IMPREF:
 preferred fw version:    02.38.00.00
 preferred carrier name:  GENERIC
 preferred config name:   GENERIC_002.082_000
 preferred subpri index:  000
 current fw version:      02.38.00.00
 current carrier name:    GENERIC
 current config name:     GENERIC_002.082_000
 current subpri index:    000

OK
AT!GOBIIMPREF?
!GOBIIMPREF:
 preferred fw version:    02.38.00.00
 preferred carrier name:  GENERIC
 preferred config name:   GENERIC_002.082_000
 preferred subpri index:  000
 current fw version:      02.38.00.00
 current carrier name:    GENERIC
 current config name:     GENERIC_002.082_000
 current subpri index:    000

OK
AT!USBSPEED?
SUPPORTED:High-Speed
CURRENT  :High-Speed

OK
AT!USBCOMP?
Config Index: 1
Config Type:  1 (Generic)
Interface bitmask: 0000100D (diag,nmea,modem,mbim)

OK
AT!USBVID?
!USBVID:
1199

OK
AT!USBPID?
!USBPID:
APP : 9079
BOOT: 9078

OK
AT!USBPRODUCT?
Sierra Wireless EM7455 Qualcomm Snapdragon X7 LTE-A

OK
AT!PRIID?
PRI Part Number: 9906870
Revision: 001.000
Customer: Lenovo-Raven2

Carrier PRI: 9999999_9904609_SWI9X30C_02.38.00.00_00_GENERIC_002.082_000

OK
AT!SELRAT?
!SELRAT: 11, UMTS and LTE Only


OK
AT!BAND?
Index, Name,                        GW Band Mask     L Band Mask      TDS Band Mask
00, All bands,                      0002000007C00000 00000100030818DF 0000000000000000


OK
AT!PCINFO?
State: Online
LPM voters - Temp:0, Volt:0, User:0, W_DISABLE:0, IMSWITCH:0, BIOS:0, LWM2M:0, OMADM:0, FOTA:0
LPM persistence - None

OK
AT!PCOFFEN?
2

OK
AT!CUSTOM?
!CUSTOM:
             GPSENABLE          0x01
             GPSSEL             0x01
             IPV6ENABLE         0x01
             SIMLPM             0x01
             USBSERIALENABLE    0x01
             FASTENUMEN         0x02
             SINGLEAPNSWITCH    0x01


OK
AT!IMAGE?
TYPE SLOT STATUS LRU FAILURES UNIQUE_ID   BUILD_ID
FW   1    GOOD   1   0 0      ?_?         02.38.00.00_?
FW   2    EMPTY  0   0 0
FW   3    EMPTY  0   0 0
FW   4    EMPTY  0   0 0
Max FW images: 4
Active FW image is at slot 1

TYPE SLOT STATUS LRU FAILURES UNIQUE_ID   BUILD_ID
PRI  FF   GOOD   0   0 0      002.082_000 02.38.00.00_GENERIC
Max PRI images: 50


OK
AT!BAND=?
Index, Name,                        GW Band Mask     L Band Mask      TDS Band Mask
00, All bands                     0002000007C00000 00000100030818DF 0000000000000000
01, Europe 3G                     0002000000400000 0000000000000000 0000000000000000
02, North America 3G              0000000004800000 0000000000000000 0000000000000000
06, Europe                        0002000000400000 00000000000800C5 0000000000000000
07, North America                 0000000004800000 000000000300185A 0000000000000000
08, WCDMA ALL                     0002000007C00000 0000000000000000 0000000000000000
09, LTE ALL                       0000000000000000 00000100030818DF 0000000000000000

                                                   0000010000000000 - B41
                                                   0000000002000000 - B26
                                                   0000000001000000 - B25
                                                   0000000000080000 - B20
                                                   0000000000001000 - B13
                                                   0000000000000800 - B12
                                                   0000000000000080 - B8
                                                   0000000000000040 - B7
                                                   0000000000000010 - B5
                                                   0000000000000008 - B4
                                                   0000000000000004 - B3
                                                   0000000000000002 - B2
                                                   0000000000000001 - B1
                                  0002000000000000 - B8  (900)
                                  0000000004000000 - B5  (850)
                                  0000000002000000 - B4 (1700)
                                  0000000001000000 - B3 (1700)
                                  0000000000800000 - B2 (1900)
                                  0000000000400000 - B1 (2100)

OK
AT!GSTATUS?
!GSTATUS:
Current Time:  170              Temperature: 32
Reset Counter: 2                Mode:        ONLINE
System mode:   LTE              PS state:    Attached
LTE band:      B7               LTE bw:      20 MHz
LTE Rx chan:   3050             LTE Tx chan: 21050
LTE CA state:  NOT ASSIGNED
EMM state:     Registered       Normal Service
RRC state:     RRC Connected
IMS reg state: No Srv

PCC RxM RSSI:  -60              RSRP (dBm):  -93
PCC RxD RSSI:  -55              RSRP (dBm):  -88
Tx Power:      8                TAC:         5F50 (24400)
RSRQ (dB):     -11.5            Cell ID:     XXXXXXXX (XXXXXXXXX)
SINR (dB):     20.0


OK
AT!LTEINFO?
!LTEINFO:
Serving:   EARFCN MCC MNC   TAC      CID Bd D U SNR PCI  RSRQ   RSRP   RSSI RXLV
             3050 466  92 24400 XXXXXXXX  7 5 5  18  22  -7.6  -88.1  -60.2  35

IntraFreq:                                          PCI  RSRQ   RSRP   RSSI RXLV
                                                     22  -7.6  -88.1  -60.2  35

InterFreq: EARFCN ThresholdLow ThresholdHi Priority PCI  RSRQ   RSRP   RSSI RXLV

WCDMA:     UARFCN ThreshL ThreshH Prio PSC   RSCP  ECN0 RXLV


OK
AT!LTECA?
ERROR
AT!LTECA=?
!LTECA=<flag>
<flag>: 0 - disable CA
        1 - enable CA


OK
AT!LTECA
ERROR
AT!LTECA=1
ERROR
AT!LTECA=0
ERROR

Thank you.

**Reply by pccr10001** [reply]
Hi. 
Thanks for replying.

I did `AT!OPENLOCK` again but `AT!LTECA` command is still not worked

```
AT!OPENLOCK?
D556C771XXXXXXXX

OK
AT!OPENLOCK="75CDD58CXXXXXXXX"
OK
AT!LTECA?
ERROR
AT!LTECA=1
OK
AT!GSTATUS?
!GSTATUS:
Current Time:  679              Temperature: 32
Reset Counter: 1                Mode:        ONLINE
System mode:   LTE              PS state:    Attached
LTE band:      B7               LTE bw:      20 MHz
LTE Rx chan:   3050             LTE Tx chan: 21050
LTE CA state:  NOT ASSIGNED
EMM state:     Registered       Normal Service
RRC state:     RRC Idle
IMS reg state: No Srv

PCC RxM RSSI:  -52              RSRP (dBm):  -79
PCC RxD RSSI:  -52              RSRP (dBm):  -78
Tx Power:      --               TAC:         5F50 (24400)
RSRQ (dB):     -6.8             Cell ID:     XXXXXXXX (XXXXXXXXX)
SINR (dB):     23.2


OK
AT!LTECA?
ERROR
AT!LTECA=1
OK
```

**Comment by nicolae788** [comment]
I see that it accepts the command which changes that setting, but it errors out when you try to query. Please enter advanced mode first AT!ENTERCND, then AT! OPENLOCK. This way query will work and setting the parameter will also work.

**Reply by pccr10001** [reply]
Hi.

Thank for replying.
I already entered advance mode command, but it didn't work.

```
AT!ENTERCND="A710"
OK
AT!OPENLOCK?
E637AAB0XXXXXXXX

OK
AT!OPENLOCK="466C89FBXXXXXXXX"
OK
AT!LTECA?
ERROR
AT!LTECA=1
OK
AT!LTECA?
ERROR
```

**Comment by nicolae788** [comment, marked as answer]
Sorry to hear that. There could be several reasons of which I am not aware. However on a simple searc I found this on the below link. Maybe you have to reflash the firmware mentioned in the post"It works well on EM7455 firmware SWI9X30C_02.33.03.00." "https://forum.sierrawireless.com/t/at-lteca-lteca-not-implemented-and-or-time-out-error/22485/2" Good luck.

**Reply by pccr10001** [reply]
Hi.
I downgraded fw to 02.33.03.00 from [here](https://ftp.opengear.com/download/cellfw/), and it's working now.

Using following files:
SWI9X30C_02.33.03.00_GENERIC_002.072_001.nvu
SWI9X30C_02.33.03.00.cwe

Thank you very much.

**Comment by nicolae788** [comment]
Great, glad it worked. Good luck in your future endeavors !
